### ISSUE TITLE
ESI Ajax Very Slow!

### ISSUE THREAD

**Issue by vitalized** [nexcess/magento-turpentine#148]
I've had a search in the issues and can't find anything about my issue... If this is a repost then, sorry!

I have Turpentine working and set the Cart & Product Compare to use ESI & Ajax.  I've been testing the page load speeds and everything loads quickly (thanks to Turpentine) except the some **random ESI blocks which take anywhere from 4 to 10 secs to load**.  It seems Turpentine is picking up some of the JS files and creating ajax ESI blocks which I haven't told it to do.  In the first screenshot Turpentine is creating ESI blocks for "powertour.settings.js".  I removed this JS file, refreshed all the caches, reloaded the page so it gets cached in Varnish and the hit the reload page to view the cached page... Now it's picking up "cloud-zoom.1.0.2.min.js" (see second screenshot)... So I removed the Cloud Zoom script and then it picked up a file called "reward.js"... I have a feeling it's going to continue like this through all my JS files.
Is there a way to stop Turpentine from creating ESI blocks for JS files?

![Screen Shot 2013-04-13 at 9 58 14 AM](https://f.cloud.github.com/assets/636206/376274/5a51ed7e-a419-11e2-80f5-e818880466fa.png)

![Screen Shot 2013-04-13 at 10 00 27 AM](https://f.cloud.github.com/assets/636206/376279/6222a12e-a419-11e2-95ea-bab4f88d5fe5.png)

**Comment by dartanandan** [comment]
Hello,

I have installed Turpentine cache everything works fine. but ajax add to cart takes long loading time. Some times add to cart takes 62 sec  or 63 sec. 

where i can find the local setup xml file, please advice me.

**Comment by aricwatson** [comment]
You can use your current theme's local.xml file to tweak things.

Do you see the same slowness if you have the bypass cookie set up (under *Cache Management* in the magento back-end)? Do you see any errors in the logs?

**Comment by dartanandan** [comment]
Hi aricwatson,

I have inserted to this xml file app/design/frontend/theme/default/layout/local.xml
Am not use bypass cookie. no error logs.

But i have the same issue. The main problem is when i change one category to another category, The first time i click add to cart button it takes long time(62 sec or 63 sec) loading. Then the issue comes some times only.

please look into my development site http://devel.tresbizz.com/nl/nl/

sorry to my bad english.

**Comment by dartanandan** [comment]
Hello,

Any one there?

**Comment by aricwatson** [comment]
I was able to replicate the issue - sometimes adding to cart is relatively fast, but once it took about a minute to complete. I did not see any similar slowness when testing with the bypass cookie present, although I did see one add to cart action take about 15 seconds. 

Have you tried adding the AJAX request URL for this action to the [URL blacklist](https://github.com/nexcess/magento-turpentine/wiki/Configuration#url-based-caching)? What cache types are active on your *cache management* screen?

**Comment by dartanandan** [comment]
I have added checkout/cart/add in url block list , and here i attached my cache management screen short.

![cache management](https://cloud.githubusercontent.com/assets/17564242/14821163/4d12f3c2-0be7-11e6-93c5-3d37a7566d06.png)

**Comment by aricwatson** [comment]
I'd suggest turning of all the cache types except for the last two (the varnish ones), then clear all the caches and re-apply the VCL. 

> How to add ajax url in block list? please explain me.

In the back-end configuration screen for Turpentine there's a *URL Blacklist* where you can add a regular expression pattern. If a request matches this pattern, it will be passed through Varnish to Magento. It's worth a try to add patter that matches your Ajax request URLs here so you can hole-punch for those.

**Comment by dartanandan** [comment]
Hi,

dartanandan@gmail.com 
please give me your email address, i will send login details. Please check and give me solution.

**Comment by aricwatson** [comment]
I'm sorry, I cannot login to non-Nexcess servers to help with Turpentine issues. 

After you added those patterns to the blacklist, did you re-apply the VCL and clear the varnish cache types? Did you try turning off all the other cache types besides the two varnish ones and test?

**Comment by aricwatson** [comment]
Closing for lack of feedback.
